types/highland/index.d.ts

```diff
@@ -1078,6 +1078,70 @@ declare namespace Highland {
 		series<U>(): Stream<U>;
 
 		/**
+		 * Transforms a stream using an arbitrary target transform.
+		 *
+		 * If `target` is a function, this transform passes the current Stream to it,
+		 * returning the result.
+		 *
+		 * If `target` is a [Duplex
+		 * Stream](https://nodejs.org/api/stream.html#stream_class_stream_duplex_1),
+		 * this transform pipes the current Stream through it. It will always return a
+		 * Highland Stream (instead of the piped to target directly as in
+		 * [pipe](#pipe)). Any errors emitted will be propagated as Highland errors.
+		 *
+		 * **TIP**: Passing a function to `through` is a good way to implement complex
+		 * reusable stream transforms. You can even construct the function dynamically
+		 * based on certain inputs. See examples below.
+		 *
+		 * @id through
+		 * @section Higher-order Streams
+		 * @name Stream.through(target)
+		 * @param {Function | Duplex Stream} target - the stream to pipe through or a
+		 * function to call.
+		 * @api public
+		 *
+		 * // This is a static complex transform.
+		 * function oddDoubler(s) {
+		 *     return s.filter(function (x) {
+		 *         return x % 2; // odd numbers only
+		 *     })
+		 *     .map(function (x) {
+		 *         return x * 2;
+		 *     });
+		 * }
+		 *
+		 * // This is a dynamically-created complex transform.
+		 * function multiplyEvens(factor) {
+		 *     return function (s) {
+		 *         return s.filter(function (x) {
+		 *             return x % 2 === 0;
+		 *         })
+		 *         .map(function (x) {
+		 *             return x * factor;
+		 *         });
+		 *     };
+		 * }
+		 *
+		 * _([1, 2, 3, 4]).through(oddDoubler); // => 2, 6
+		 *
+		 * _([1, 2, 3, 4]).through(multiplyEvens(5)); // => 10, 20
+		 *
+		 * // Can also be used with Node Through Streams
+		 * _(filenames).through(jsonParser).map(function (obj) {
+		 *     // ...
+		 * });
+		 *
+		 * // All errors will be propagated as Highland errors
+		 * _(['zz{"a": 1}']).through(jsonParser).errors(function (err) {
+		 *   console.log(err); // => SyntaxError: Unexpected token z
+		 * });
+		 */
+		through<R, U>(f: (x: R) => U): Stream<U>;
+		//TODO: don't just use `any`, find a way to preserve types with the ReadWriteStream
```

Comment:
This would only be possible if `NodeJS.ReadWriteStream` were generic right? I'm wondering if leaving a comment here is useful since nothing can be done on this side. What do you think?

Reply:
Agree; it's probably covered by the todo on [L15](https://github.com/r-k-b/DefinitelyTyped/blob/d390c327df85981ea12a303b4fe84f929b79dce6/types/highland/index.d.ts#L15):
> use externalised Readable/Writable (not node's)